DynamicTablesPkg/Library/Common/AmlLib/CodeGen/AmlResourceDataCodeGen.c

```diff
@@ -1475,6 +1475,78 @@ AmlCodeGenRdRegister (
   return LinkRdNode (RdNode, NameOpNode, NewRdNode);
 }
 
+/** Code generation for the "IO ()" ASL function.
+
+  The Resource Data effectively created is a IO Resource
+  Data. Cf ACPI 6.5:
+   - s19.6.65 IO (IO Resource Descriptor Macro)
+   - s6.4.2.5 I/O Port Descriptor
+
+  The created resource data node can be:
+   - appended to the list of resource data elements of the NameOpNode.
+     In such case NameOpNode must be defined by a the "Name ()" ASL statement
+     and initially contain a "ResourceTemplate ()".
+   - returned through the NewRdNode parameter.
+
+  @param [in]  IsDecoder16          Decoder parameter.
+                                    TRUE if 16-bit decoder.
+                                    FALSE if 10-bit decoder.
+  @param [in]  AddressMinimum       Minimum address.
+  @param [in]  AddressMaximum       Maximum address.
+  @param [in]  Alignment            Alignment.
+  @param [in]  RangeLength          Range length.
+  @param [in]  NameOpNode           NameOp object node defining a named object.
+                                    If provided, append the new resource data
+                                    node to the list of resource data elements
+                                    of this node.
+  @param [out] NewRdNode            If provided and success,
+                                    contain the created node.
+
+  @retval EFI_SUCCESS               The function completed successfully.
+  @retval EFI_INVALID_PARAMETER     Invalid parameter.
+**/
+EFI_STATUS
+EFIAPI
+AmlCodeGenRdIo (
+  IN  BOOLEAN IsDecoder16,
+  IN  UINT16 AddressMinimum,
+  IN  UINT16 AddressMaximum,
+  IN  UINT8 Alignment,
+  IN  UINT8 RangeLength,
+  IN  AML_OBJECT_NODE_HANDLE NameOpNode, OPTIONAL
+  OUT AML_DATA_NODE_HANDLE    *NewRdNode  OPTIONAL
+  )
+{
+  EFI_STATUS                   Status;
+  EFI_ACPI_IO_PORT_DESCRIPTOR  IoDesc;
+  AML_DATA_NODE                *IoNode;
+
+  if (AddressMinimum > AddressMaximum) {
+    return EFI_INVALID_PARAMETER;
+  }
+
+  IoDesc.Header.Byte = ACPI_IO_PORT_DESCRIPTOR;
+  IoDesc.Information = IsDecoder16 ? BIT0 : 0;
+
+  IoDesc.BaseAddressMin = AddressMinimum;

```

Comment:
Hello Abdul,
NIT:
```
AddressMin evaluates to a 16-bit integer that specifies the minimum acceptable starting address for the I/O range. It
must be an even multiple of AddressAlignment. [...]
AddressMax evaluates to a 16-bit integer that specifies the maximum acceptable starting address for the I/O range. It
must be an even multiple of AddressAlignment. [...]
```
Would it be possible to check the min/max addresses are correct regarding the alignment ?
Otherwise lgtm

Reply:
Hi Pierre Gondois,
  Thanks for the quick review. I have addressed the review comments.
Please review the updated patch.
Thanks
AbduL